### ISSUE TITLE
[BUG] `RuntimeError: protection exception (16777921)`

### ISSUE THREAD

**Issue by kr1tx** [dashingsoft/pyarmor#1274]
Packed script doesn't work with `restrict_module = 2` after updating pyarmor to 8.2.1 and updated pyinstaller to 5.11.0

```
Traceback (most recent call last):
  File "<frozen __main__>", line 3, in <module>
  File "<frozen Program>", line 5, in <module>
RuntimeError: protection exception (16777921)
[14524] Failed to execute script 'Program' due to unhandled exception!

```

**Comment by jondy** [comment]
Could you provide a simple script and options to reproduce this issue?

**Comment by kr1tx** [comment]
I did not face this issue with the pyinstaller version 5.9.0 and pyarmor version 8.1
Here is a simple script and options to reproduce this issue
```
pyinstaller --upx-dir=C:\upx  -F Program.py -i fuse.ico
pyarmor gen -e 2023-07-20 --mix-str --enable-bcc --assert-call --enable-jit --assert-import --restrict --pack dist\Program.exe Program.py
```
Program.py:
```
import requests
print(requests.get('https://google.com'))
```
ruler file:
```
[builder]
restrict_module = 2
```

**Comment by kr1tx** [comment]
`restrict_module = 3` also did not work

**Comment by jondy** [comment]
How about remove `--enable-bcc`? It may not work with `--restrict`

**Comment by kr1tx** [comment]
it worked be4 updating

**Comment by kr1tx** [comment]
waiting for the update

**Comment by jondy** [comment]
Fixed in 8.2.2
But now doesn't use option `--restrict` (or restrict_module=3) with `--pack`, it doesn't work.
Only use `--private` or `restrict_module=2`

**Comment by kr1tx** [comment]
Thanks
